python/tskit/trees.py

```diff
@@ -5442,3 +5442,50 @@ def newick_trees(self, precision=3, breakpoints=None, Ne=1):
             "This method is no longer supported. Please use the Tree.newick"
             " method instead"
         )
+
+    def nexus(self, precision=14, node_labels=None):
+        """
+        Returns a nexus encoding <https://en.wikipedia.org/wiki/Nexus_file>`_
+        of this tree.
+        :param int precision: The numerical precision with which branch lengths are
+            printed.
+        :param dict node_labels: If specified, show custom labels for the nodes
+            that are present in the map. Any nodes not specified in the map will
+            not have a node label.
+        :return: A newick representation of this tree.
+        :rtype: str
+        """
+        sample_tree = self.first()
```

Comment:
Things are bit simpler if we use [samples](https://tskit.readthedocs.io/en/latest/data-model.html#definitions) rather than leaves. Then, we have 
```
node_labels = {u: str(u) for u in self.samples()}
```
then we don't need to look at the first tree at all.